CHANGELOG.md

```diff
@@ -3,6 +3,7 @@
 ### Enhancements
 
 * Throws an IllegalArgumentExcpetion when calling Realm.copyToRealm()/Realm.copyToRealmOrUpdate() with a RealmObject which belongs to another Realm instance in a different thread.
+* Uses ArrayList to store references to make finalizer faster (#2496).
```

Comment:
This sounds really internal to me. Perhaps just "Improved speed when cleaning up native resources" or something like that. At least something that the user better can relate to?